tiledb/sm/array/array.cc

```diff
@@ -1560,5 +1629,97 @@ void Array::set_serialized_array_open() {
   is_open_ = true;
 }
 
+uint64_t Array::max_tile_size() {
+  tiledb_fragment_max_tile_sizes_t max_tile_sizes;
+
+  if (remote_) {
+    auto rest_client = storage_manager_->rest_client();
+    if (rest_client == nullptr) {
+      throw StatusException(
+          Status_Error("Failed to obtain maximum tile size; remote array with "
+                       "no REST client."));
+    }
+
+    throw_if_not_ok(rest_client->get_array_maximum_tile_size_from_rest(
+        array_uri_, &max_tile_sizes.max_in_memory_tile_size));
+    return max_tile_sizes.max_in_memory_tile_size;
+  }
+
+  // Check that array is open
+  if (!is_open()) {
+    throw StatusException(Status_ArrayError(
+        "[Array::array_get_fragment_tile_max_size] Array is closed"));
+  }
+
+  if (query_type_ != QueryType::READ) {
+    throw StatusException(Status_ArrayError(
+        "[Array::array_get_fragment_tile_max_size] needs QueryType::READ"));
+  }
+
+  EncryptionKey enc_key(config_);
+  uint64_t timestamp_start = 0;
+  uint64_t timestamp_end = UINT64_MAX;
+
+  // Create an ArrayDirectory object and then load the array schemas
+  // and fragment metadata.
+  auto array_dir =
+      ArrayDirectory(resources_, array_uri_, timestamp_start, timestamp_end);
+
+  MemoryTracker memory_tracker;
+  bool found = false;
+  uint64_t memory_budget;
+  throw_if_not_ok(
+      config_.get<uint64_t>("sm.mem.total_budget", &memory_budget, &found));
+  assert(found);
+  // Set the memory budget for the array
+  if (!memory_tracker.set_budget(memory_budget)) {
+    throw StatusException(
+        Status_ArrayError("Memory budget is too small to open array"));
+  }
+
+  auto&& [st, array_schema_latest, array_schemas_all, fragment_metadata] =
+      storage_manager_->load_array_schemas_and_fragment_metadata(
+          array_dir, &memory_tracker, enc_key, true);
```

Comment:
I think we want to leave out vacuumed fragments from this calculation on principle. We're computing maximum tile sizes in preparation for fixed-size data block use. Data inside a fragment to be vacuumed will never need to be loaded into memory.

Reply:
@eric-hughes-tiledb A fragment to be vacuumed might still be loaded if the user time travels to timestamps that do not include the consolidated fragment that is referenced by the .vac file.